core/src/main/clojure/xtdb/indexer/content_log_indexer.clj

```diff
@@ -0,0 +1,114 @@
+(ns xtdb.indexer.content-log-indexer
+  (:require
+   [juxt.clojars-mirrors.integrant.core :as ig]
+   [xtdb.blocks :as blocks]
+   [xtdb.types :as types]
+   [xtdb.util :as util]
+   [xtdb.vector.writer :as vw])
+  (:import
+   (java.io Closeable)
+   (java.util ArrayList)
+   (java.util.concurrent CompletableFuture)
+   (java.util.concurrent.atomic AtomicInteger)
+   (java.util.function Consumer)
+   (org.apache.arrow.memory BufferAllocator)
+   (org.apache.arrow.vector VectorLoader VectorSchemaRoot VectorUnloader)
+   (org.apache.arrow.vector.types.pojo Schema)
+   (xtdb ICursor)
+   (xtdb.object_store ObjectStore)))
+
+#_{:clj-kondo/ignore [:unused-binding :clojure-lsp/unused-public-var]}
+(definterface IContentOpIndexer
+  (^void logPut [^String table-name, ^org.apache.arrow.vector.ValueVector doc-rel, ^long doc-offset])
+  ;; (^void logPut [^String table-name, ^xtdb.vector.IIndirectRelation doc-rel, ^long doc-offset])
+  (^void commit [])
+  (^void abort []))
+
+#_{:clj-kondo/ignore [:unused-binding :clojure-lsp/unused-public-var]}
+(definterface IContentIndexer
+  (^xtdb.indexer.content_log_indexer.IContentOpIndexer startTx [])
+  (^void finishBlock [])
+  (^java.util.concurrent.CompletableFuture finishChunk [^long chunkIdx])
+  (^void nextChunk [])
+  (^void close []))
+
+(defn- ->log-obj-key [chunk-idx]
+  (format "chunk-%s/content-log.arrow" (util/->lex-hex-string chunk-idx)))
+
+(defmethod ig/prep-key :xtdb.indexer/content-log-indexer [_ opts]
+  (merge {:allocator (ig/ref :xtdb/allocator)
+          :object-store (ig/ref :xtdb/object-store)}
+         opts))
+
+(def content-log-schema
+  (Schema. [(types/->field "documents" types/dense-union-type false)]))
```

Comment:
@jarohen The schema here is:
- documents (duv)
  - table-name (duv-leg name)
    - documents for that table also a `duv`
  
One could potentially also add the tables as top-level vectors to the schema.

Reply:
Anyways one could potentially lay it out a bit differently. (`StructVector` with unions pushed down etc...)